File: arrays/java/Array3.java
import java.util.*;

class Array3{
    public static void main(String[] args){
        int[] arr1={1,1,2,3,4,5};
	    int[] arr2={2,3,4,4,5,6};
	    unionOfArray(arr1, arr2);
    }

//Union using two Pointer method.
    static void unionOfArray(int[] arr1, int[] arr2){
	    int n1=arr1.length;
	    int n2=arr2.length;
	    int j=0;
	    int i=0;
        List<Integer> union = new ArrayList<>();
        while(i<n1 && j<n2){
            if(arr1[i]<=arr2[j]){
                // int lastElement = union.get(union.size()-1);
                if(union.size()==0 || union.get(union.size()-1)!=arr1[i]) union.add(arr1[i]);
                i++;
            }else{
                // int lastElement = union.get(union.size()-1);
                if(union.size()==0 || union.get(union.size()-1)!=arr2[j]) union.add(arr2[j]);
                j++;
            }
        }
        while(i<n1){
            // int lastElement = union.get(union.size()-1);
                if(union.size()==0 || union.get(union.size()-1)!=arr1[i]) union.add(arr1[i]);
                i++;
        }
        while(j<n2){
            // int lastElement = union.get(union.size()-1);
                if(union.size()==0 || union.get(union.size()-1)!=arr2[j]) union.add(arr2[j]);
                j++;
        }
        System.out.println(union); 

    }
}

File: arrays/csharp/Array3.cs
public class Array3{
    public static void Main(string []args){
        int[] arr1={1,1,2,3,4,5};
	    int[] arr2={2,3,4,4,5,6};
	    UnionOfArray(arr1, arr2);
    }

    void UnionOfArray(int[] arr1,int[] arr2){
        int n1 = arr1.Length;
        int n2 = arr2.Length;
        List<int> union = new List<int>();
        int i=0;
        int j=0;
        while(i<n1 && j<n2){
            if(arr1[i]<=arr2[j]){
                if(union.Count==0 || union[union.Count-1]!=arr1[i]) union.Add(arr1[i]);
                i++;
            }else{
                if(union.Count==0 || union[union.Count-1]!=arr2[j]) union.Add(arr2[j]);
                j++;
            }
        }
        while(i<n1){
            if(union.Count==0 || union[union.Count-1]!=arr1[i]) union.Add(arr1[i]);
            i++;
        }
        while(j<n2){
            if(union.Count==0 || union[union.Count-1]!=arr2[j]) union.Add(arr2[j]);
            j++;
        }
    }

}